src/emulator/CMakeLists.txt

```diff
@@ -1,10 +1,28 @@
 set(VITA3K_ORG_NAME "Vita3K")
 set(VITA3K_APP_NAME "Vita3K")
 
-set(VITA3K_VERSION $ENV{TRAVIS_BRANCH})
+if(APPLE)
+	set(VITA3K_VERSION $ENV{TRAVIS_BRANCH})
+elseif(WIN32)
+	set(VITA3K_VERSION "")
 
-if(VITA3K_VERSION STREQUAL "")
-	set(VITA3K_VERSION "development version")
+	if(VITA3K_VERSION STREQUAL "")
+		execute_process(
+			COMMAND git rev-list HEAD --count
+			WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+			OUTPUT_VARIABLE GIT_COUNT
+			OUTPUT_STRIP_TRAILING_WHITESPACE)
+		execute_process(
+			COMMAND git rev-parse --short HEAD
+			WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+			OUTPUT_VARIABLE GIT_HASH
+			OUTPUT_STRIP_TRAILING_WHITESPACE)
+			if(GIT_HASH STREQUAL "")
+				set(VITA3K_VERSION "Development version")
+			else()
+				set(VITA3K_VERSION "v0.1 ${GIT_COUNT}-${GIT_HASH}")
```

Comment:
I think I'd prefer the version numbers to come from CI rather than manual git inspection.

https://www.appveyor.com/docs/environment-variables/
`APPVEYOR` exists and is true when building via AppVeyor.
`APPVEYOR_REPO_TAG_NAME` says the tag name. If that's undefined then `APPVEYOR_REPO_BRANCH` says the branch.
`APPVEYOR_BUILD_NUMBER` says the build number, but I'm not sure why we care.
`APPVEYOR_REPO_COMMIT` says the commit hash, but again, I don't think we need to display it.

Similarly for Travis:
https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables
`TRAVIS` is set to true
`TRAVIS_BRANCH` is the branch name, or the tag name when the build is a tag (release.)
`TRAVIS_BUILD_NUMBER` is the build number, but I don't think we need to display it.
I think `TRAVIS_COMMIT` is the commit hash, but I don't think we need to display it.

So I think I'd probably do something like this:

`set(VITA3K_VERSION "development version")`
`if AppVeyor is set`
`set VITA3K_VERSION from tag if it's set, branch if it's not`
`else if travis is set`
`set VITA3K_VERSION from TRAVIS_BRANCH`

Reply:
Appveyor code is no needed, like i have say, appveyor build works like this and fix regression for git version, and show exactly same think me with compile myself Visual Studio.
For travis mac, no idea, if you can test, but normally is same think travis

Reply:
OK, thanks for the explanation. I appreciate the preference for having one solution to the problem that works on both AppVeyor and Travis.

Would it be possible to get the version number from the current git tag? Something like `git describe --abbrev=0 --tags` should work.

I'm not sure what that returns if the current commit doesn't have a tag.

We should be able to get rid of the `if(APPLE)` and `if(VITA3K_VERSION STREQUAL "")` checks then, and just call execute_process from the outermost level.

Thanks again for this.

Reply:
no, i have tested "git describe --abbrev=0 --tags" on my git and don't works like you see so is bad
![capture d ecran 2018-02-05 19 38 48](https://user-images.githubusercontent.com/5261759/35822331-3f245c7c-0aac-11e8-8fbd-a910d194ff4d.png)

And for other, i don't know, if you can test my code for see if works, and after teste without If WIN32 and if APPLE for see if works same for you.
but for "if(VITA3K_VERSION STREQUAL "")" io think is obligatory, for user not have git or user gitwindows only like that return Devlopement version" and no empty